api/type.rb

```diff
@@ -200,9 +200,18 @@ def initialize(name)
       NAME = Api::Type::String.new('name')
     end
 
+    # Properties that are fetched externally
+    class FetchedExternal < Type
+      attr_writer :resource
+
+      def api_name
+        name
+      end
+    end
+
     # Represents a fingerprint.  A fingerprint is an output-only
     # field used for optimistic locking during updates.
-    class Fingerprint < String
+    class Fingerprint < FetchedExternal
```

Comment:
Hm, what does that do to the Terraform output code?

Reply:
This shouldn't do anything.

Reply:
Great!